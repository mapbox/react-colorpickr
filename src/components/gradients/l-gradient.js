'use strict';

import React from 'react';
import PropTypes from 'prop-types';
import themeable from 'react-themeable';
import { autokey } from '../../autokey';

class LGradient extends React.PureComponent {
  static propTypes = {
    theme: PropTypes.object.isRequired,
    active: PropTypes.bool.isRequired,
    opacityLow: PropTypes.number.isRequired,
    opacityHigh: PropTypes.number.isRequired
  };

  render() {
    const theme = autokey(themeable(this.props.theme));
    const { active, opacityLow, opacityHigh } = this.props;
    if (!active) return <noscript />;

    // Opacity should be 0 when range value is at 0.5
    const high = (opacityHigh - 0.5) * 2;
    const low = (opacityLow - 0.5) * 2;

    return (
      <div>
        <div {...theme('gradient', 'gradientLight')} />
        <div 
          {...theme('gradient')}
          style={{
            background: 'linear-gradient(to bottom, rgba(255,255,255,0) 0%, rgb(128,128,128) 100%)',
          }}
        />
        <div
          {...theme('gradient')}
          style={{
            background: 'rgb(255,255,255)',
            opacity: high
          }}
        />
        <div
          {...theme('gradient')}
          style={{
            background: 'rgb(0,0,0)',
            opacity: low
          }}
        />
      </div>
    );
  }
}

export default LGradient;
