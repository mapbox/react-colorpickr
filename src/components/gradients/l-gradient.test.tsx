import React from 'react';
import { render } from '@testing-library/react';
import { LGradient } from './l-gradient';

describe('LGradient', () => {
  test('renders inactive', () => {
    const props = {
      active: false,
      theme: {},
      opacityLow: 0.5,
      opacityHigh: 0.6
    };

    const { baseElement } = render(<LGradient {...props} />);
    expect(baseElement).toMatchSnapshot();
  });

  test('renders active', () => {
    const props = {
      active: true,
      theme: {},
      opacityLow: 0,
      opacityHigh: 0
    };

    const { baseElement } = render(<LGradient {...props} />);
    expect(baseElement).toMatchSnapshot();
  });
});
