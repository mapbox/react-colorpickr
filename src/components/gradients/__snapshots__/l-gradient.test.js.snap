// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LGradient renders active 1`] = `
<body>
  <div>
    <div />
    <div />
    <div
      style="background: rgb(255, 255, 255); opacity: -1;"
    />
    <div
      style="background: rgb(0, 0, 0); opacity: -1;"
    />
  </div>
</body>
`;

exports[`LGradient renders inactive 1`] = `
<body>
  <div>
    <noscript />
  </div>
</body>
`;
